gapic/src/main/com/google/gapid/util/Crash2ExceptionHandler.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gapid.util;
+
+import com.google.gapid.Version;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.MultipartBuilder;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+import okio.Buffer;
+
+/**
+ * Handles uncaught exceptions and sends stacktraces to Crash2 server.
+ */
+public class Crash2ExceptionHandler implements Thread.UncaughtExceptionHandler {
+  private static final String CRASH_REPORT_PRODUCT = "GAPID";
+  private static final String CRASH_REPORT_VERSION = "Client:" + Version.GAPID_VERSION.toString();
+  // TODO(baldwinn860): Send to production url when we get approval.
+  private static final String CRASH_REPORT_URL_BASE = "https://clients2.google.com/cr/staging_report?";
+  private static final String CRASH_REPORT_URL_FIELDS = "product=" + CRASH_REPORT_PRODUCT + "&version=" + CRASH_REPORT_VERSION;
+  private static final String CRASH_REPORT_URL = CRASH_REPORT_URL_BASE + CRASH_REPORT_URL_FIELDS;
+  private static final Logger LOG = Logger.getLogger(Crash2ExceptionHandler.class.getName());
+
+  private Thread.UncaughtExceptionHandler oldDefaultHandler_;
+  public Crash2ExceptionHandler(Thread.UncaughtExceptionHandler oldDefaultHandler) {
+    oldDefaultHandler_ = oldDefaultHandler;
+  }
+
+  @Override
+  public void uncaughtException(Thread t, Throwable th) {
+    ByteArrayOutputStream traceStream = new ByteArrayOutputStream();
+    th.printStackTrace(new PrintStream(traceStream));
+    String traceString = traceStream.toString();
+
+    try {
+      // Creates a connection to crash2
+      OkHttpClient client = new OkHttpClient();
+      RequestBody body = new MultipartBuilder()
+        .type(MultipartBuilder.FORM)
+        .addFormDataPart("product", CRASH_REPORT_PRODUCT)
+        .addFormDataPart("version", CRASH_REPORT_VERSION)
+        .addFormDataPart("exception_info", traceString)
+        .build();
+      Request request = new Request.Builder()
+        .url(CRASH_REPORT_URL)
+        .post(body)
```

Comment:
Since you are not using the body anywhere else, you can inline the builder - c.f. proto builders.